### PR TITLE
embed: explicitly define unix proto

### PR DESCRIPTION
**Pull request by hexfusion**
This is pending https://github.com/grpc/grpc-go/pull/1883

ref: https://github.com/coreos/etcd/issues/9340

### CODE REVIEW

**Comment by hexfusion** [comment]
Upstream grpc-go fixes will be merged next week allowing unix support to work.

**Comment by hexfusion** [comment]
> WARNING: 2018/04/10 10:53:42 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp: address unix://localhost:15452000000: too many colons in address"; Reconnecting to {unix://localhost:15452000000 0  <nil>}

I need to understand how this can happen, seems like tests are not using patched grpc-go.

**Comment by gyuho** [comment]
@hexfusion etcd master still uses old gRPC, so we may see it for awhile :0 Also we need to wait until we upgrade gRPC in master branch including your fix on unix socket.

File: embed/serve.go [hunk 1]
```diff
@@ -194,7 +195,15 @@ type registerHandlerFunc func(context.Context, *gw.ServeMux, *grpc.ClientConn) e
 
 func (sctx *serveCtx) registerGateway(opts []grpc.DialOption) (*gw.ServeMux, error) {
 	ctx := sctx.ctx
-	conn, err := grpc.DialContext(ctx, sctx.addr, opts...)
+
+	addr := sctx.addr
+	// explictly define unix network for gRPC socket support
```
Comment (gyuho) [inline embed/serve.go]: Let's make it simpler?

```go
if sctx.network == "unix" {
    addr = fmt.Sprintf("%s://%s", network, addr)
}
```

Comment (hexfusion) [reply embed/serve.go]: sure, all set

Comment (gyuho) [reply embed/serve.go]: lgtm thanks @hexfusion !

**Pull request merged**